imagepuller/internal/service/verify.go

```diff
@@ -8,67 +8,80 @@ import (
 	"errors"
 	"fmt"
 	"log/slog"
-	"strings"
 
 	"github.com/google/go-containerregistry/pkg/name"
-	v1 "github.com/google/go-containerregistry/pkg/v1"
+	gcr "github.com/google/go-containerregistry/pkg/v1"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
 	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/google/go-containerregistry/pkg/v1/types"
 )
 
-func (s *ImagePullerService) getAndVerifyImage(ctx context.Context, log *slog.Logger, imageURL string) (v1.Image, error) {
+// Error types which should be differentiable in tests.
+var (
+	ErrParseDigest   = errors.New("parsing image digest")
+	ErrRemoteImage   = errors.New("obtaining remote image")
+	ErrRemoteIndex   = errors.New("obtaining remote image index")
+	ErrValidateLayer = errors.New("validating layer")
+)
+
+func (s *ImagePullerService) getAndVerifyImage(ctx context.Context, log *slog.Logger, imageURL string) (gcr.Image, error) {
 	ref, err := name.NewDigest(imageURL)
 	if err != nil {
-		return nil, fmt.Errorf("parsing image digest: %w", err)
+		return nil, fmt.Errorf("%w: %w", ErrParseDigest, err)
 	}
 
 	tr := transport.NewRetry(remote.DefaultTransport)
 
-	remoteImgIndex, err := remote.Index(ref, remote.WithContext(ctx), remote.WithTransport(tr))
-	if isDigestMismatch(err) {
-		return nil, fmt.Errorf("validating image ref: %w", err)
+	desc, err := remote.Head(ref, remote.WithContext(ctx), remote.WithTransport(tr))
```

Comment:
You could consider adding support for HEAD requests to the registry.

Reply:
Pretty sure it already does - the unittest PR I am working on  is based on this branch, and I apparently am getting a valid response, since [these test cases](https://github.com/edgelesssys/contrast/blob/97288573ab8ba6974c66c25e7dc0fb90bcdbe3bf/imagepuller/internal/service/verify_test.go#L159) pass.

Reply:
The registry currently responds to a HEAD request as it would to a GET request - with a response payload. Seems to be fine for the library, I just wanted to mention it for completeness' sake.

Reply:
Ah, got it. Do you think it's worth it then?